scripts/gb-model/create_region_shapes.py

```diff
@@ -0,0 +1,421 @@
+                                        )
+                                        new_polygons.append(polygon)
+                                else:
+                                    # No actual split occurred
+                                    new_polygons.append(polygon)
+                            else:
+                                # Single geometry result - no split
+                                new_polygons.append(polygon)
+
+                        except Exception as split_error:
+                            logger.warning(
+                                f"Split failed for boundary {boundary_idx}: {split_error}"
+                            )
+                            new_polygons.append(polygon)
+                    else:
+                        # No intersection - keep original polygon
+                        new_polygons.append(polygon)
+
+                # Update current polygons for next iteration
+                current_polygons = new_polygons
+
+            logger.info(
+                f"Country {idx} split into {len(current_polygons)} regions using {split_count} boundaries"
+            )
+
+            # Create region entries from final polygons
+            for i, polygon in enumerate(current_polygons):
+                if polygon.is_valid and not polygon.is_empty:
+                    # Create region data
+                    region_data = country.copy()
+                    region_data["geometry"] = polygon
+                    region_data["region_id"] = f"region_{region_id:03d}"
+                    region_data["original_country_id"] = idx
+                    region_data["sub_region_id"] = i
+                    region_data["area_km2"] = polygon.area / 1000000  # Convert to km²
+                    region_data["num_boundaries_used"] = split_count
+
+                    regions.append(region_data)
+                    region_id += 1
+
+        except Exception as e:
+            logger.error(f"Error processing country {idx}: {e}")
+            # Fallback: keep original geometry
+            region_data = country.copy()
+            region_data["region_id"] = f"region_{region_id:03d}"
+            region_data["original_country_id"] = idx
+            region_data["sub_region_id"] = 0
+            region_data["area_km2"] = country.geometry.area / 1000000
+            region_data["num_boundaries_used"] = 0
+            regions.append(region_data)
+            region_id += 1
+
+    # Create GeoDataFrame from regions
+    if regions:
+        regions_gdf = gpd.GeoDataFrame(regions, crs=country_shapes.crs)
+        logger.info(
+            f"Successfully created {len(regions_gdf)} regions from {len(country_shapes)} original shapes"
+        )
+        return regions_gdf
+    else:
+        logger.error("No regions were created!")
+        return country_shapes.copy()
+
+
+def clean_regions(regions_gdf, min_area_threshold=1000):
```

Comment:
probably something more concrete would be good 
```suggestion
def drop_small_regions(regions_gdf, min_area_threshold=1000):
```